kubernetes-ingress/templates/controller-deployment.yaml

```diff
@@ -178,9 +178,16 @@ spec:
             valueFrom:
               fieldRef:
                 fieldPath: metadata.namespace
-          {{- range .Values.controller.extraEnvs }}
-          - name: {{ .name }}
-            value: {{ .value }}
+          {{- if .Values.controller.extraEnvs -}}
+          {{- toYaml .Values.controller.extraEnvs | nindent 10 }}
+{{/*      {{- range .Values.controller.extraEnvs }}*/}}
+{{/*      - name: {{ .name }}*/}}
+{{/*        {{- if hasKey . "value" }}*/}}
+{{/*        value: {{ .value }}*/}}
+{{/*        {{- else if hasKey . "valueFrom" }}*/}}
+{{/*        valueFrom: {{- toYaml .valueFrom | nindent 14 }}*/}}
+{{/*        {{- end }}*/}}
+{{/*      {{- end }}*/}}
```

Comment:
@Mo3m3n @dkorunic as you guys can see I left the "alternative" approach to doing this commented out but committed here in the diff.

The two ways of doing this are:

1. Trust users and just blindly copy-paste `.Values.controller.extraEnvs`; or
2. Check for specific keys like `value` and `valueFrom` and only template those into the controller deployment.

I actually think the first option (just trust your chart users; worst case scenario Helm template will error, right?) is most appropriate and avoids having to replicate the syntax validation that Helm does for you anyway.

Reply:
Let me know which you prefer and I'll update this accordingly.

Reply:
I would say that this first approach is sensible and works, let's keep that as is. Thanks @mecampbellsoup!